### PR TITLE
[CI] Migrate GraphTrainer & RL H100 integration tests to OSDC (ARC)

### PR DESCRIPTION
**Pull request by huydhn**
Migrate the two remaining `linux.aws.h100.8` integration tests to the OSDC (ARC) H100 runners (`mt-l-bx86iamx-176-1800-h100-8`), following the same pattern as #3464, which migrated `integration_test_8gpu_h100.yaml`.

### Workflows migrated
- `.github/workflows/integration_test_8gpu_graph_trainer_h100.yaml`
- `.github/workflows/integration_test_8gpu_rl_h100.yaml`

### Changes (both files)
- Route through `set-matrix.yaml` with `runner-cuda: mt-l-bx86iamx-176-1800-h100-8` and `is-experimental: true` (CUDA-only, no ROCm) — matching the other experiment workflows (`integration_test_4gpu_rl`, `8gpu_torchft`, `8gpu_transformers_modeling_backend`).
- `linux_job_v2.yml` → `linux_job_v3.yml`.
- Pull the private-ECR image tagged with the `.ci/docker` tree hash that `set-matrix.yaml` returns, instead of the bare public image name.
- Install torch from `${{ matrix.index-url }}` and clear `PIP_EXTRA_INDEX_URL` so the in-cluster cpu pypi-cache can't supply a `+cpu` torch build (FA3's `test/cu130` index in the RL workflow is preserved as-is).
- Drop `sudo mkdir/chown` for the artifacts dir (the v3 container user uid matches the OSDC runner); fix the `permissions:` block indentation.

### Behavior note
As with the sibling experiment workflows, the CUDA job runs on push to `main`, on the cron schedule, and on PR `synchronize` while the relevant ciflow label (`ciflow/h100.8` / `ciflow/rl`) is present. The bare `labeled` event itself produces an empty matrix (a property of the shared `set-matrix.yaml` + `is-experimental` path), so a subsequent push triggers the actual run.

### Motivation
`linux.aws.h100.8` capacity is being consolidated onto OSDC/ARC H100 runners; this finishes the migration started in #3464.

### CODE REVIEW

File: .github/workflows/integration_test_8gpu_rl_h100.yaml
Comment (wwwjn) [inline .github/workflows/integration_test_8gpu_rl_h100.yaml:1]: CI failure becuase of `uv pip isntall -e .` failed, this seems be true. We will need install torchtitan because monarch / vllm spawn child process and we'll need something like `import torchtitan.config` work

**Comment by SherlockNoMad** [comment]
Thanks for moving the CI to OSDC pool. 

If you see some test case failing in GraphTrainer CI, you can feel free to land, and leave the test case fixing to us, as long as you have runner picking up the job and env setup.

**Review by SherlockNoMad — APPROVED** [review]

File: .ci/docker/common/install_deepep.sh [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
-set -ex
+set -eux
 
 DEEPEP_COMMIT=${DEEPEP_COMMIT:-1b8f467}
 
 # Dependencies for DeepEP compilation (NVSHMEM needs libcudacxx, IB headers).
 sudo apt-get update -qq && sudo apt-get install -y -qq rdma-core libibverbs1 libmlx5-1 libibverbs-dev
 sudo apt-get autoclean && sudo apt-get clean
-sudo rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+sudo rm -rf /var/lib/apt/lists/* /var/tmp/*
```
Comment (huydhn) [inline .ci/docker/common/install_deepep.sh:20]: @SherlockNoMad I found the line responsible for https://github.com/pytorch/torchtitan/actions/runs/26801406725/job/79008692892. https://github.com/pytorch/torchtitan/blob/main/.ci/docker/common/install_deepep.sh#L20 has a step to `rm -rf /tmp/*` and it unknowingly deletes the CI logs used by the runner there.

Removing that part with a tracking issue to fix that properly on the runner https://github.com/pytorch/ci-infra/issues/690

**Comment by huydhn** [comment]
> Thanks for moving the CI to OSDC pool.
> 
> If you see some test case failing in GraphTrainer CI, you can feel free to land, and leave the test case fixing to us, as long as you have runner picking up the job and env setup.

The failed GraphTrainer tests FYI https://github.com/pytorch/torchtitan/actions/runs/26802927029/job/79013498849
